code/datums/mutations/body.dm

```diff
@@ -336,15 +336,16 @@
 	text_gain_indication = "<span class='userdanger'>A horrible burning sensation envelops you as your flesh turns to acid!</span>"
 	text_lose_indication = "<span class='notice'>A feeling of relief fills you as your flesh goes back to normal.</span>"
 	difficulty = 18//high so it's hard to unlock and use on others
-	var/msgcooldown = 0
+	/// The cooldown for the warning message
+	COOLDOWN_DECLARE(msgcooldown)
 
-/datum/mutation/human/acidflesh/on_life()
-	if(prob(25))
-		if(world.time > msgcooldown)
+/datum/mutation/human/acidflesh/on_life(delta_time, times_fired)
+	if(DT_PROB(13, delta_time))
+		if(COOLDOWN_FINISHED(src, msgcooldown))
 			to_chat(owner, "<span class='danger'>Your acid flesh bubbles...</span>")
-			msgcooldown = world.time + 200
-		if(prob(15))
-			owner.acid_act(rand(30,50), 10)
+			COOLDOWN_START(src, msgcooldown, 20 SECONDS)
+		if(DT_PROB(7.5, delta_time))
```

Comment:
Forgive me if my memory is going bust, but I recall something about nested probs like this only needing the first one converted, is that incorrect?

Reply:
You are correct. I'll fix that.